core/controller/src/main/scala/whisk/core/controller/WebActions.scala

```diff
@@ -216,32 +216,48 @@ protected[core] object WhiskWebActionsApi extends Directives {
                 case _ => throw new Throwable("Invalid header")
             } getOrElse List()
 
+            val body = fields.get("body")
+
             val code = fields.get(rp.statusCode).map {
                 case JsNumber(c) =>
                     // the following throws an exception if the code is
                     // not a whole number or a valid code
                     StatusCode.int2StatusCode(c.toIntExact)
 
                 case _ => throw new Throwable("Illegal code")
-            } getOrElse (OK)
+            }
+
+            body map {
+                case JsString(str) =>
```

Comment:
Would it make sense to add the `str.nonEmpty` as a guard and let the rest fall through to the default case which is identical to the `else` case here?

Reply:
Thinking about it, we could simplify it even more maybe:

```scala
body.collect {
    case JsString(str) if str.nonEmpty => 
        interpretHttpResponse(code.getOrElse(OK), headers, str, transid)
    case js if js != JsNull => 
        interpretHttpResponseAsJson(code.getOrElse(OK), headers, js, transid)
}.getOrElse(respondWithEmptyEntity(code.getOrElse(NoContent), headers))

Should be equivalent to the existing code.

Reply:
After the first match the second will succeed. Since the type is JsString of an empty string. You need a second check.

Reply:
```
            body.collect {
                case JsString(str) if str.nonEmpty   => interpretHttpResponse(code.getOrElse(OK), headers, str, transid)
                case JsString(str) /* str.isEmpty */ => respondWithEmptyEntity(code.getOrElse(NoContent), headers)
                case js if js != JsNull              => interpretHttpResponseAsJson(code.getOrElse(OK), headers, js, transid)
            } getOrElse respondWithEmptyEntity(code.getOrElse(NoContent), headers)
```

an alternative.

Reply:
Meh. You're right of course. Worth it? (The last version?)